File: app/src/main/java/com/sumian/sleepdoctor/main/WelcomeActivity.java
package com.sumian.sleepdoctor.main;

import android.view.View;

import com.blankj.utilcode.util.LogUtils;
import com.sumian.sleepdoctor.R;
import com.sumian.sleepdoctor.account.bean.Token;
import com.sumian.sleepdoctor.account.login.LoginActivity;
import com.sumian.sleepdoctor.account.userProfile.activity.ImproveUserProfileOneActivity;
import com.sumian.sleepdoctor.app.AppManager;
import com.sumian.sleepdoctor.base.BaseActivity;
import com.sumian.sleepdoctor.leancloud.LeanCloudManager;

/**
 * Created by jzz
 * on 2018/1/16.
 * desc:
 */

public class WelcomeActivity extends BaseActivity {

    public static final int WELCOME_SHOW_TIME = 500;

    @Override
    protected int getLayoutId() {
        return R.layout.activity_main_welcome;
    }

    @Override
    protected void initWidget(View root) {
        super.initWidget(root);
        setStatusBar();
    }

    @Override
    protected void initData() {
        super.initData();
        AppManager.getAccountViewModel().getLiveDataToken().observe(this, token -> {
            LogUtils.d(token);
            onToken(token);
        });
    }

    public void onToken(Token token) {
        mRoot.postDelayed(() -> {
            if (token == null) {
                LoginActivity.show(WelcomeActivity.this, LoginActivity.class);
            } else if (token.is_new) {
                ImproveUserProfileOneActivity.show(WelcomeActivity.this, ImproveUserProfileOneActivity.class);
            } else {
                MainActivity.show(WelcomeActivity.this, MainActivity.class);
            }
            finish();
        }, WELCOME_SHOW_TIME);
        if (token != null) {
            LeanCloudManager.getAndUploadCurrentInstallation();
        }
    }
}

File: doctor/src/main/java/com/sumian/sddoctor/welcome/WelcomeActivity.kt
package com.sumian.sddoctor.welcome

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.Observer
import com.blankj.utilcode.util.ActivityUtils
import com.sumian.sddoctor.R
import com.sumian.sddoctor.account.TokenInfo
import com.sumian.sddoctor.app.AppManager
import com.sumian.sddoctor.login.login.LoginActivity
import com.sumian.sddoctor.main.MainActivity
import com.sumian.sddoctor.util.SumianExecutor

class WelcomeActivity : AppCompatActivity() {

    companion object {
        private const val WELCOME_SHOW_TIME = 500L
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_welcome)
        AppManager.getAccountViewModel().getTokenInfo()
                .observe(this, Observer<TokenInfo> { t -> onTokenInfoChange(t) })
    }

    private fun onTokenInfoChange(tokenInfo: TokenInfo?) {
        SumianExecutor.runOnUiThread({
            if (tokenInfo != null && !tokenInfo.isExpired()) {
                ActivityUtils.startActivity(MainActivity::class.java)
            } else {
                ActivityUtils.startActivity(LoginActivity::class.java)
            }
            finish()
        }, WELCOME_SHOW_TIME)
    }
}
